plugins/modules/firewalld.py

```diff
@@ -875,6 +877,13 @@ def main():
         if changed is True:
             msgs.append("Changed icmp-block-inversion %s to %s" % (icmp_block_inversion, desired_state))
 
+        # Type of icmp_block_inversion will be changed to boolean in a future release.
+        try:
+            boolean(icmp_block_inversion, True)
+        except TypeError:
+            warnings.append('The value of icmp_block_inversion is "%s", It should be set boolean '
+                            'and will be changed from str to bool in a future release.' % icmp_block_inversion)
```

Comment:
```suggestion
            module.warn('The value of the icmp_block_inversion option is "%s". '
                        'The type of the option will be changed from string to boolean in a future release. '
                        'To avoid unexpected behavior, please change the value to boolean.' % icmp_block_inversion)
```

Reply:
done!